lib/fluent/plugin/out_split.rb

```diff
@@ -44,9 +50,18 @@ def split_message(message)
       data = {}
       message.split(@sep_regex).each do |e|
         matched = @format_regex.match(e) or next
-        data.store(matched['key'], matched['value'])
+        @store_fun.call(data, matched['key'], matched['value'])
```

Comment:
Will performance remains correct ?
My reasoning was to assign things on configuration step (called once), then just call functions during processing (called multiple times).